src/main/java/spark/SparkBase.java

```diff
@@ -241,6 +249,38 @@ public static synchronized void externalStaticFileLocation(String externalFolder
     }
 
     /**
+     * Maps the given path to the given WebSocket handler.
+     * <p>
+     * This is currently only available in the embedded server mode.
+     *
+     * @param path the WebSocket path.
+     * @param handler the handler class that will manage the WebSocket connection to the given path.
+     */
+    public static synchronized void webSocket(String path, Class<?> handler) {
```

Comment:
WebSocket handler classes can be arbitrary classes. Users can:

* Implement the `WebSocketListener` interface (or extend the convenient `WebSocketAdapter`).
* Use an arbitrary class and use the websocket annotations, as seen in [the examples](https://github.com/nacx/spark/blob/ws/src/test/java/spark/examples/websocket/EchoWebSocket.java).

That's why I left it wihtout restrictions, but I'll add a check to require the annotation or the listener interface.